src/components/ModalAuth.tsx

```diff
@@ -0,0 +1,214 @@
+import React, { ChangeEvent, FC, useCallback, useEffect, useRef, useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { alert, login } from '../action-creators'
+import { FIREBASE_REDIRECT_URL } from '../constants'
+import { ActionButton } from './ActionButton'
+import Modal from './Modal'
+
+const firebaseErrorsIndex = {
+  'auth/weak-password': 'Password should be at least 6 characters',
+  'auth/email-already-in-use': 'Account already present. Please log in',
+  'auth/wrong-password': 'Invalid username or password',
+  'auth/user-not-found': 'User not found',
+  'auth/too-many-requests': 'Account temporarily disabled. You can restore it by resetting your password',
+  default: 'Something went wront'
+}
+
+type errorCode = keyof typeof firebaseErrorsIndex
+
+type SubmitAction = (closeModal: () => void, email: string, password?: string) => Promise<void>
+
+interface InputProps {
+  type: 'email' | 'password',
+  placeholder: string,
+  onBlur: (e: ChangeEvent<HTMLInputElement>) => void,
+  value: string,
+}
+/**
+ *
+ */
+const Input:FC<InputProps> = ({ type, placeholder, value, onBlur }) => {
+  const [inputValue, updateInputValue] = useState(value)
+
+  useEffect(() => {
+    updateInputValue(value)
+  }, [value])
+
+  /** On input change handler. */
+  const onChange = (e: ChangeEvent<HTMLInputElement>) => updateInputValue(e.target.value)
+
+  return <input type={type} placeholder={placeholder} value={inputValue} onChange={onChange} onBlur={onBlur}/>
+}
```

Comment:
What's the purpose of this `Input` component? It looks like a simple wrapper for `<input>`.

Reply:
It's to prevent the re-renders of the entire modal component while the user is typing.

Reply:
Okay, thanks